tests/integration/targets/win_acl/tasks/main.yml

```
@@ -16,6 +16,13 @@
   - absent
   - present
 
+- name: add PKI module to Powershell to create testing certificate
+  win_shell: Import-Module PKI
+
+- name: create certificate for testing
+  win_shell: '(New-SelfSignedCertificate -Subject "ACL Test" -KeyExportPolicy Exportable).Thumbprint'
+  register: test_acl_certificate_thumbprint
```

It would be good to also test CryptoAPI based keys in our tests. You should be able to use use `-Provider "Microsoft Base Cryptographic Provider v1.0"` to create one of those keys

```suggestion
- name: create certificates for testing
  win_shell: |
    $certParams = @{
        KeyAlgorithm = 'RSA'
        KeyExportPolicy = 'Exportable'
        KeyLength = 2048
    }
    (New-SelfSignedCertificate @certParams -Subject "ACL Test CNG" -Provider "Microsoft Software Key Storage Provider").Thumbprint
    (New-SelfSignedCertificate @certParams -Subject "ACL Test CryptoAPI" -Provider "Microsoft Base Cryptographic Provider v1.0" -KeySpec Signature).Thumbprint
  register: test_acl_cert_info

- set_fact:
    test_acl_certificiate_cng_thumbprint: '{{ test_acl_cert_info.stdout_lines[0] }}'
    test_acl_certificiate_cryptoapi_thumbprint: '{{ test_acl_cert_info.stdout_lines[1] }}'
```

You would need to adjust the cleanup and tests to use these new values but you probably only need a single test for the Crypto stuff. No need to do all the idempotency tests for both.

---

this is a great point. definitely overlooked that. (and thanks for reminding me I can use `set_fact`! Definitely not an ansible expert😅)